enterprise/backup/src/main/java/org/neo4j/backup/BackupStrategyWrapper.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.backup;
+
+import java.io.File;
+
+import org.neo4j.commandline.admin.CommandFailed;
+import org.neo4j.helpers.OptionalHostnamePort;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.lifecycle.LifeSupport;
+
+class BackupStrategyWrapper
+{
+    private final BackupStrategyInterface backupStrategyImplementation;
```

Comment:
And this field (and associated parameter) doesn't really need to say `Implementation`.

Reply:
Addressed in next push